corehq/apps/hqadmin/views/data.py

```diff
@@ -119,7 +119,12 @@ def download_blob(request):
     """Pairs with the get_download_url utility and command"""
     key = request.GET.get("key")
     try:
-        meta = BlobMeta.objects.partitioned_get(partition_value=key, key=key)
+        meta = BlobMeta.objects.partitioned_get(
+            domain='__system__',
+            type_code=CODES.tempfile,
+            partition_value=key,
+            key=key,
+        )
```

Comment:
I don't think we have anything in the UI that lets them access blobs, so you'd probably have to go through the console.  I did initially hope to make this a generic "download any blob" view so it could be used for debugging other stuff (it'd be analogous to `raw_doc`).  However, this actually isn't generic, as it only works if `key` and `parent_id` (the `partition_value`) are the same, which is not typical, so I do think it's reasonable to be more explicit about how this is intended to be used, to avoid confusion.

Reply:
In terms of generic downloads, I would not be against listing attached blobs in the raw docs view with the ability to download from there.

Reply:
thanks @esoergel 

in order to limit the access to blobs, did we consider just giving a temporary token stored in cache, to the user that they would use along with their credentials, that is linked to the relevant blob only? Using `__system__` seems open ended.

Reply:
@mkangia I think for superuser-only access it shouldn't be necessary, though [I did consider something along those lines](https://docs.google.com/document/d/1L6CLWa69-OiWpW_NF2vKKXH2xKiISis8bfMyyvpiKtU/edit#bookmark=id.7wzm4catr8pq) to grant access to non superusers.

Here are the things I'd like to see if they become relevant, [from here](https://docs.google.com/document/d/1L6CLWa69-OiWpW_NF2vKKXH2xKiISis8bfMyyvpiKtU/edit#heading=h.gq2nh3t5lgz2):

> * Add an optional “self-destructing payload” flag, where the download view deletes the blob after it’s been accessed once
> * Make it available to non-admin users: store usernames in the blob meta properties dict, then add a non-admin view that lets those usernames download the file

Can you elaborate on what you mean by `__system__` seeming open-ended?  Since this is intended for things like management command artifacts, `domain` often doesn't apply, as many management commands collect data across domains (eg auditing feature flag usage, or logging apps that need special care to allow a migration).

Reply:
Hey @esoergel 
I simply meant that currently we are letting someone download any system generated blob file in favor of only one file they ideally should have access to, hence open ended. An access token valid for a single file only, removes the need to worry about any filter at all.

Reply:
Doesn't the unique ID of the blob already serve the same purpose as an access token?  It's an unguessable string of characters that is needed to access the file (though technically `uuid4()` isn't cryptographically secure).

I don't believe that this:
```http://localhost:8000/hq/admin/download_blob/?key=321c2430-2fd3-4b71-9f78-90a2d432a9ac&token=b19a7e7b-45f9-4b5b-b8f6-3f0b219ebbaf```
offers any more security than this:
```http://localhost:8000/hq/admin/download_blob/?key=321c2430-2fd3-4b71-9f78-90a2d432a9ac```
unless you distributed the access token separately over email or something

> currently we are letting someone download any system generated blob file in favor of only one file they ideally should have access to

This is only available to superusers at present, who already have access to `raw_doc` and others, and they'd still need the ID to be able to retrieve it.

The purpose of this PR (adding `domain` and `type_code` to the `get()`) is not to limit the blobs that can be accessed through this view for security, but to further codify the intention of this view, because it wouldn't work as a generic blob downloader anyways.  For what it's worth, I'd be in support of taking this the other direction too - turning this into a utility that downloads arbitrary blobs.

Reply:
Thanks for the response @esoergel 

> I don't believe that this:
http://localhost:8000/hq/admin/download_blob/?key=321c2430-2fd3-4b71-9f78-90a2d432a9ac&token=b19a7e7b-45f9-4b5b-b8f6-3f0b219ebbaf
offers any more security than this:
http://localhost:8000/hq/admin/download_blob/?key=321c2430-2fd3-4b71-9f78-90a2d432a9ac
unless you distributed the access token separately over email or something

You are right. 

> not to limit the blobs that can be accessed through this view for security

my bad. I thought there was originally a security concern.

Reply:
Right, I came to this PR from the same slack thread. :) 
But jumped straight to the changes without reading the description.